Sources/BraintreeCoreSwift/BTGraphQLHTTP.swift

```diff
@@ -0,0 +1,244 @@
+import Foundation
+
+// TODO: Make internal once rest of core is in Swift
+@objcMembers public class BTGraphQLHTTP: BTHTTP {
+
+    public typealias RequestCompletion = (BTJSON?, HTTPURLResponse?, Error?) -> Void
+
+    // MARK: - Properties
+
+    private let exceptionName: NSExceptionName = NSExceptionName("")
+
+    // MARK: - Overrides
+
+    public override func get(_ path: String, completion: @escaping RequestCompletion) {
+        NSException(name: exceptionName, reason: "GET is unsupported").raise()
+    }
+
+    public override func get(_ path: String, parameters: NSDictionary? = nil, completion: RequestCompletion?) {
+        NSException(name: exceptionName, reason: "GET is unsupported").raise()
+    }
+
+    public override func post(_ path: String, completion: @escaping RequestCompletion) {
+        httpRequest(method: "POST", parameters: nil, completion: completion)
+    }
+
+    public override func post(_ path: String, parameters: NSDictionary? = nil, completion: @escaping RequestCompletion) {
+        httpRequest(method: "POST", parameters: parameters, completion: completion)
+    }
+
+    public override func put(_ path: String, completion: @escaping RequestCompletion) {
+        NSException(name: exceptionName, reason: "PUT is unsupported").raise()
+    }
+
+    public override func put(_ path: String, parameters: NSDictionary? = nil, completion: RequestCompletion?) {
+        NSException(name: exceptionName, reason: "PUT is unsupported").raise()
+    }
+
+    public override func delete(_ path: String, completion: @escaping RequestCompletion) {
+        NSException(name: exceptionName, reason: "DELETE is unsupported").raise()
+    }
+
+    public override func delete(_ path: String, parameters: NSDictionary? = nil, completion: RequestCompletion?) {
+        NSException(name: exceptionName, reason: "DELETE is unsupported").raise()
+    }
+
+    // MARK: - Internal methods
+    
+    func httpRequest(
+        method: String,
+        parameters: NSDictionary? = [:],
+        completion: RequestCompletion?
+    ) {
+        var errorUserInfo: [String: Any] = [:]
+
+        if self.baseURL.absoluteString.isEmpty || self.baseURL.absoluteString == "" {
+            errorUserInfo["method"] = method
+            errorUserInfo["parameters"] = parameters
+            let error = Self.constructError(code: .missingBaseURL, userInfo: errorUserInfo)
```

Comment:
Do we have plans to convert this to something like ....

```
let error = BTHTTPError.missingBaseURL(userInfo)
```

like we're using for some other `Error.swift` files?

Reply:
Right now `BTHTTPError` is also used in `BTCardClient`, `PayPalClient`, `ThreeDSecure`, and `UnionPay` modules. We likely can update it in those places, but that's a larger piece of work. There is already a ticket for this work so it's certainly something we should tackle in next major version (ticket ends in `1898`)

Reply:
Added a TODO for this here: https://github.com/braintree/braintree_ios/blob/876a040bf14922b9875b4f1c3d12936a57c47ee7/Sources/BraintreeCoreSwift/BTHTTP.swift#L490